sdks/python/apache_beam/io/gcp/bigquery_io_read_pipeline.py

```diff
@@ -14,66 +14,135 @@
+from apache_beam import Map
+from apache_beam import ParDo
+from apache_beam.io import BigQueryDisposition
+from apache_beam.io import BigQuerySource
+from apache_beam.io import Read
+from apache_beam.io import WriteToBigQuery
+from apache_beam.io.gcp.bigquery_tools import BigQueryWrapper
+from apache_beam.io.gcp.bigquery_tools import parse_table_schema_from_json
+from apache_beam.testing.load_tests.load_test import LoadTest
+from apache_beam.testing.synthetic_pipeline import SyntheticSource
 from apache_beam.testing.test_pipeline import TestPipeline
 from apache_beam.testing.util import assert_that
 from apache_beam.testing.util import equal_to
-
-
-class RowToStringWithSlowDown(beam.DoFn):
-
-  def process(self, element, num_slow=0, *args, **kwargs):
-
-    if num_slow == 0:
-      yield ['row']
-    else:
-      rand = random.random() * 100
-      if rand < num_slow:
-        time.sleep(0.01)
-        yield ['slow_row']
-      else:
+from apache_beam.transforms.combiners import Count
+
+
+class BigQueryIOReadTest(LoadTest):
+  def setUp(self):
+    super(BigQueryIOReadTest, self).setUp()
+    self.num_slow = self.pipeline.get_option('num_slow') or 0
+    self.input_dataset = self.pipeline.get_option('input_dataset')
+    self.input_table = self.pipeline.get_option('input_table')
+    self._check_for_input_data()
+
+  def _check_for_input_data(self):
+    """Checks if a BQ table with input data exists and creates it if not."""
+    wrapper = BigQueryWrapper()
+    try:
+      wrapper.get_table(self.project_id, self.input_dataset, self.input_table)
+    except Exception:
+      self._create_input_data()
+
+  def _create_input_data(self):
+    """
+    Runs an additional pipeline which creates test data and waits for its
+    completion.
+    """
+    SCHEMA = parse_table_schema_from_json(
+        '{"fields": [{"name": "data", "type": "BYTES"}]}')
+
+    def format_record(record):
+      # Since Syntentic Source returns data as a dictionary. we should skip
+      # one of the part
+      return {'data': base64.b64encode(record[1])}
+
+    p = TestPipeline()
+    # pylint: disable=expression-not-assigned
+    (p
+     | 'Produce rows' >> Read(SyntheticSource(self.parseTestPipelineOptions()))
+     | 'Format' >> Map(format_record)
+     | 'Write to BigQuery' >> WriteToBigQuery(
+         self.input_dataset + '.' + self.input_table,
+         schema=SCHEMA,
+         create_disposition=BigQueryDisposition.CREATE_IF_NEEDED,
+         write_disposition=BigQueryDisposition.WRITE_EMPTY))
+    p.run().wait_until_finish()
+
+  class RowToStringWithSlowDown(DoFn):
+    def process(self, element, num_slow=0, *args, **kwargs):
+      if num_slow == 0:
         yield ['row']
+      else:
+        rand = random.random() * 100
+        if rand < num_slow:
+          time.sleep(0.01)
+          yield ['slow_row']
+        else:
+          yield ['row']
 
+  def test(self):
+    run_pipeline(self.pipeline, self.input_dataset, self.input_table,
```

Comment:
I am also not sure if the slow reading was meant to be reading from SyntheticSources.

Reply:
Correct me if I'm wrong, but I think the slow reading doesn't care where do the data come from. Its purpose is to just slow down the execution.